### DISCUSSION TITLE
组合API中使用onBeforeRouteUpdate和onBeforeRouteLeave触发时机

### DISCUSSION THREAD

**Discussion by cismylove** [vuejs/router#1837] [Help and Questions] [answered]
### Reproduction

none

### Steps to reproduce the bug

```js
import { onBeforeRouteUpdate,onBeforeRouteLeave } from 'vue-router';

// 不同级的路由不能触发,如/operation/gift/material=>/log/index
onBeforeRouteUpdate(to => {
	setTags(to);
});

// 同级别路由不能触发,如/operation/ToppingRoom=>/operation/sticker/material
onBeforeRouteLeave((to,from)=>{
	// console.log(to,'onBeforeRouteLeave');
	// console.log(from,'onBeforeRouteLeave');
	setTags(to);
})
```

### Expected behavior

路由改变了，就能触发导航守卫

### Actual behavior

onBeforeRouteUpdate在形如/operation/gift/material=>/log/index的跨级别路由不能触发守卫
onBeforeRouteLeave在形如/operation/gift/material=>/operation/sticker/material的同一个父路由下的不同子路由不能触发守卫

**Comment by posva** [comment, marked as answer]
This is the intended behavior of these guards: https://router.vuejs.org/zh/guide/advanced/navigation-guards.html#%E5%8F%AF%E7%94%A8%E7%9A%84%E9%85%8D%E7%BD%AE-api

You can use a global navigation guard instead `router.beforeResolve()` or an after hook `router.afterEach()`
